tests/test_types.py

```diff
@@ -202,6 +204,189 @@ class ConListModel(BaseModel):
     ]
 
 
+def test_conlist():
+    class Model(BaseModel):
+        foo: List[int] = Field(..., min_items=2, max_items=4)
+        bar: conlist(str, min_items=1, max_items=4) = None
+
+    assert Model(foo=[1, 2], bar=['spoon']).dict() == {'foo': [1, 2], 'bar': ['spoon']}
+
+    with pytest.raises(ValidationError, match='ensure this value has at least 2 items'):
+        Model(foo=[1])
+
+    with pytest.raises(ValidationError, match='ensure this value has at most 4 items'):
+        Model(foo=list(range(5)))
+
+    assert Model.schema() == {
+        'title': 'Model',
+        'type': 'object',
+        'properties': {
+            'foo': {'title': 'Foo', 'type': 'array', 'items': {'type': 'integer'}, 'minItems': 2, 'maxItems': 4},
+            'bar': {'title': 'Bar', 'type': 'array', 'items': {'type': 'string'}, 'minItems': 1, 'maxItems': 4},
+        },
+        'required': ['foo'],
+    }
+
+    with pytest.raises(ValidationError) as exc_info:
+        Model(foo=[1, 'x', 'y'])
+    assert exc_info.value.errors() == [
+        {'loc': ('foo', 1), 'msg': 'value is not a valid integer', 'type': 'type_error.integer'},
+        {'loc': ('foo', 2), 'msg': 'value is not a valid integer', 'type': 'type_error.integer'},
+    ]
+
+    with pytest.raises(ValidationError) as exc_info:
+        Model(foo=1)
+    assert exc_info.value.errors() == [{'loc': ('foo',), 'msg': 'value is not a valid list', 'type': 'type_error.list'}]
+
+
+def test_constrained_set_good():
+    class ConSetModelMax(BaseModel):
+        v: conset(int) = []
+
+    m = ConSetModelMax(v=[1, 2, 3])
+    assert m.v == {1, 2, 3}
+
+
+def test_constrained_set_default():
+    class ConSetModelMax(BaseModel):
+        v: conset(int) = {}
```

Comment:
This test falils on my side indicating that is not `set`, but rather a `dict`.

Reply:
Matter is failing, maybe that's why?

Reply:
Matter?

Reply:
Sorry, master.

Reply:
This is very weird, as per #1682 this can be fixed by setting the default to a set.

However this should still pass since we don't have `validate_always=True` - the default value (an empty dict) should be okay.

The question is: what has changed on master since this was started that's causing this to fail when merged, and could that be a smoke signal to a bigger problem somewhere?

Reply:
validate_always is getting set to `True` as per 8aebba2c35007b848aceed2f8ae55fd7b4888ab6

Reply:
Yes I just saw that. I was investigating, then had to stop. I'll try to look later.